### PR TITLE
feat(cli): sort composed resources in the render output

### LINKED ISSUE #5008: Sort `crossplane beta render` output
### What problem are you facing?

The current behavior of Crossplane beta render output is causing challenges during local debugging. Specifically, the output is not sorted, making it harder for developers to analyze and understand the rendered resources.

### How could Crossplane help solve your problem?
The render output should be sorted, preferably by the 'kind' field, to provide a more organized and easily understandable view for local debugging.

The current workaround involves using external tools such as Python to manually sort the output, as demonstrated by the following command:

```
python3 -c "import yaml; data = list(yaml.load_all(open('$file'), Loader=yaml.FullLoader)); data.sort(key=lambda x: x.get('kind', '')); yaml.dump_all(data, open('$file', 'w'), default_flow_style=False)"
```

Implement native sorting within the Crossplane CLI for the render output, either by default or through a command-line option.

### PR DESCRIPTION
**Pull request by phisco**
### Description of your changes

Fixes #5008 by sorting composed resources by composed resource name (`crossplane.io/composition-resource-name` annotation) to ensure the output is stable.

I have: 

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added or updated unit tests.
- [ ] ~Added or updated e2e tests.~
- [ ] ~Linked a PR or a [docs tracking issue] to [document this change].~
- [ ] ~Added `backport release-x.y` labels to auto-backport this PR.~

Need help with this checklist? See the [cheat sheet].

[contribution process]: https://github.com/crossplane/crossplane/tree/master/contributing
[docs tracking issue]: https://github.com/crossplane/docs/issues/new
[document this change]: https://docs.crossplane.io/contribute/contribute
[cheat sheet]: https://github.com/crossplane/crossplane/tree/master/contributing#checklist-cheat-sheet

### CODE REVIEW

**Review by lsviben — APPROVED** [review]

**Comment by phisco** [comment]
@jbw976 I didn't add the backport label here as I don't see this as a "bug" in the current implementation, but I could be easily convinced otherwise, let me know what you think.

File: cmd/crank/beta/render/cmd.go [hunk 1]
```diff
@@ -206,3 +211,13 @@ func (c *Cmd) Run(k *kong.Context, _ logging.Logger) error { //nolint:gocyclo //
...
 	return nil
 }
+
+// sortByKindAndName sorts a slice of composed resources in place, first by GroupVersionKind and then by name.
+func sortComposedResources(out []composed.Unstructured) {
+	sort.Slice(out, func(i, j int) bool {
+		if comp := strings.Compare(out[i].GroupVersionKind().String(), out[j].GroupVersionKind().String()); comp != 0 {
+			return comp < 0
+		}
+		return strings.Compare(out[i].GetName(), out[j].GetName()) < 0
```
Comment (negz) [inline cmd/crank/beta/render/cmd.go:221]: I notice the godoc for `strings.Compare` says:

> Compare is included only for symmetry with package bytes. It is usually clearer and always faster to use the built-in string comparison operators ==, <, >, and so on.

```suggestion
		return out[i].GroupVersionKind.String() + out[i].GetName() < out[j].GroupVersionKind.String() + out[j].GetName()
```

I'd find this more intuitive to read, personally.

**Comment by phisco** [comment]
now I got why @haarchri, was talking about sorting by kind AND annotations... most of the times resources won't have a name as `metadata.name`. I'll switch to sorting it by the resource name instead.

**Review by negz — APPROVED** [review]
Thanks @phisco!

**Pull request merged**